pkg/cmd/roachtest/tests/backup_fixtures.go

```diff
@@ -229,179 +231,139 @@ func (bd *backupDriver) monitorBackups(ctx context.Context) {
-			initWorkloadViaRestore: &restoreSpecs{
-				backup:                 backupSpecs{version: "v22.2.1", numBackupsInChain: 48},
-				restoreUptoIncremental: 48,
-			},
-			// Use weekly to allow an over 24 hour timeout.
+			fixture: MediumFixture,
+			hardware: makeHardwareSpecs(hardwareSpecs{
+				workloadNode: true,
+				nodes:        9,
+				cpus:         16,
+			}),
+			timeout: 24 * time.Hour,
 			suites:  registry.Suites(registry.Weekly),
-			timeout: 48 * time.Hour,
-			skip:    "only for fixture generation",
+			clouds:  []spec.Cloud{spec.AWS, spec.GCE},
 		},
 		{
-			hardware: makeHardwareSpecs(hardwareSpecs{workloadNode: true}),
-			scheduledBackupSpecs: makeBackupFixtureSpecs(scheduledBackupSpecs{
-				backupSpecs: backupSpecs{
-					workload:           tpccRestore{opts: tpccRestoreOptions{warehouses: 5000}},
-					nonRevisionHistory: true,
-				},
+			fixture: LargeFixture,
+			hardware: makeHardwareSpecs(hardwareSpecs{
+				workloadNode: true,
+				nodes:        9,
+				cpus:         32,
+				volumeSize:   4000,
 			}),
-			initWorkloadViaRestore: &restoreSpecs{
-				backup:                 backupSpecs{version: "v23.1.1", numBackupsInChain: 48},
-				restoreUptoIncremental: 48,
-			},
-			timeout: 1 * time.Hour,
-			suites:  registry.Suites(registry.Nightly),
-			skip:    "only for fixture generation",
+			timeout: 24 * time.Hour,
+			suites:  registry.Suites(registry.Weekly),
+			// The large fixture is only generated on GCE to reduce the cost of
+			// storing the fixtures.
+			clouds: []spec.Cloud{spec.GCE},
 		},
-	} {
+	}
+	for _, bf := range specs {
 		bf := bf
-		bf.initTestName()
+		clusterSpec := bf.hardware.makeClusterSpecs(r)
 		r.Add(registry.TestSpec{
-			Name:              bf.testName,
+			Name: fmt.Sprintf(
+				"backupFixture/tpcc/warehouses=%d/incrementals=%d",
+				bf.fixture.ImportWarehouses, bf.fixture.IncrementalChainLength,
+			),
 			Owner:             registry.OwnerDisasterRecovery,
-			Cluster:           bf.hardware.makeClusterSpecs(r, bf.scheduledBackupSpecs.cloud),
+			Cluster:           clusterSpec,
 			Timeout:           bf.timeout,
 			EncryptionSupport: registry.EncryptionMetamorphic,
-			CompatibleClouds:  bf.scheduledBackupSpecs.CompatibleClouds(),
+			CompatibleClouds:  registry.Clouds(bf.clouds...),
 			Suites:            bf.suites,
 			Skip:              bf.skip,
 			Run: func(ctx context.Context, t test.Test, c cluster.Cluster) {
-
-				bd := makeBackupDriver(t, c, bf)
+				registry := newFixtureRegistry(ctx, t, c)
+
+				// Piggy back on fixture generation to run the GC. Run the GC first to
+				// bound the number of fixtures leaked if fixture creation is broken.
+				require.NoError(t, registry.GC(ctx, t.L()))
+
+				handle, err := registry.Create(ctx, bf.fixture.Name, t.L())
+				require.NoError(t, err)
+
+				bd := backupDriver{
+					t:        t,
+					c:        c,
+					sp:       bf,
+					version:  clusterupgrade.CurrentVersion(),
```

Comment:
I think your change could allow us to remove the manual binary version upload logic and remove the `version` field from the backupDriver. Previously, i needed to create a 24.1 fixture on a roachtest running on master. I don't think we need this with your change.